sympy/polys/polytools.py

```diff
@@ -4551,6 +4570,32 @@ def degree_list(f, *gens, **args):
 
     return tuple(map(Integer, degrees))
 
+@public
+def power_list(f, *gens, **args):
+    """
+    Return a list of powers of ``f`` in all variables.
+
+    Examples
+    ========
+
+    >>> from sympy import power_list
+    >>> from sympy.abc import x, y
+
+    >>> power_list(x**2 + y*x + 1)
+    (2, 1, 0)
+
```

Comment:
More examples. May be corner cases.

Reply:
How will this feature respond to, `x**8 * (1 - x)**9`? Similarly for large values like, `x**100000 * (1 - x)**10000000`?

Reply:
>May be corner cases.

Added now.
>How will this feature respond to, x**8 * (1 - x)**9?

Responds as it is expected to respond.
```python
>>> power_list(x**8 * (1 - x)**9)
(17, 16, 15, 14, 13, 12, 11, 10, 9, 8)
>>> expand(x**8 * (1 - x)**9)
-x**17 + 9*x**16 - 36*x**15 + 84*x**14 - 126*x**13 + 126*x**12 - 84*x**11 + 36*x**10 - 9*x**9 + x**8
```
Also this test is added in test file.

Reply:
feel free to resolve this conversation